libs/ui/design-system/src/lib/components/ShowMoreText/ShowMoreText.tsx

```diff
@@ -0,0 +1,84 @@
+import classNames from "classnames";
+import { CSSProperties, useCallback, useEffect, useRef, useState } from "react";
+
+import "./ShowMoreText.scss";
+
+const CLASS_NAME = "amp-show-more-text";
+
+export type Props = {
+  gapAfterText?: number;
+  lineClamp?: number;
+  children: React.ReactNode;
+};
+
+export const ShowMoreText = ({
+  children,
+  gapAfterText = 20,
+  lineClamp = 3,
+}: Props) => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const [open, setOpen] = useState(false);
+  const [showLink, setShowLink] = useState(false);
+  const [textHeight, setTextHeight] = useState<number>(0);
+  const [shortHeight, setShortHeight] = useState<number>(0);
+
+  const handleShowMoreClick = useCallback(
+    (e: React.MouseEvent<HTMLSpanElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      setOpen(true);
+      ref?.current && setTextHeight(ref.current.scrollHeight + gapAfterText);
+    },
+    [ref]
+  );
+
+  const handleShowLessClick = useCallback(
+    (e: React.MouseEvent<HTMLSpanElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      setOpen(false);
+      setTextHeight(shortHeight);
+    },
+    [shortHeight]
+  );
+
+  useEffect(() => {
+    if (ref.current && ref.current.clientHeight < ref.current.scrollHeight) {
```

Comment:
this should be calculated by the text when the invoke the component. so it will be one line or up to 3 lines.